hr_timesheet_holiday/models/hr_holidays.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Sunflower IT <http://sunflowerweb.nl>
+# Copyright 2019 Coop IT Easy SCRLfs
+#   - Vincent Van Rossem <vincent@coopiteasy.be>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import datetime
+from datetime import timedelta
+
+
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning as UserError
+
+
+class HrHolidays(models.Model):
+    """Update analytic lines on status change of Leave Request"""
+
+    _inherit = "hr.holidays"
+
+    analytic_line_ids = fields.One2many(
+        "account.analytic.line", "leave_id", "Analytic Lines"
+    )
+
+    @api.multi
+    def add_timesheet_line(self, description, date, hours, account):
+        """Add a timesheet line for this leave"""
+        self.ensure_one()
+        # User exists because already check during the holidays_validate
+        user = self.employee_id.user_id
+        self.sudo().with_context(force_write=True).write(
+            {
+                "analytic_line_ids": [
+                    (
+                        0,
+                        False,
+                        {
+                            "name": description,
+                            "date": date,
+                            "unit_amount": hours,
+                            "company_id": self.employee_id.company_id.id,
+                            "account_id": account.id,
+                            "user_id": user.id,
+                            "is_timesheet": True,
+                        },
+                    )
+                ]
+            }
+        )
+
+    @api.model
+    def _get_company_hours_per_day(self, company):
+        hours_per_day = company.timesheet_hours_per_day
+        if not hours_per_day:
+            raise UserError(
+                _("No hours per day defined for Company '%s'")
```

Comment:
"No hours" should be singular. prefere "Hours per day not defined for company '%s'"

Reply:
done